src/backend/api/reservations.js

```diff
@@ -0,0 +1,98 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// let knexQuery = knex("reservations");
+
+//************ Returns all reservations ************
+
+router.get("/", async (request, response) => {
+  try {
+    // const reservations = await knex("reservations");
+    // response.json(reservations);
+
+    const checkIfAvailable = await knex("meals")
+      .join("reservations", "meals.id", "reservations.meal_id")
+      .select(
+        "reservations.meal_id",
+        "meals.id",
+        "meals.title",
+        "meals.max_reservations",
+        "reservations.number_of_guests"
+      )
+      .sum("reservations.number_of_guests")
+      .groupBy("meals.id");
+    // .where("meals.id", "reservations.meal_id");
+    response.json(checkIfAvailable);
+  } catch (error) {
+    throw error;
+  }
+});
```

Comment:
I think what your hw wants you to do is use an availableReservations param in the get meals request, just like the rest of them (maxPrice, title etc..). So, you should move this over there :)

Also, does this query work for you? I'm getting an error.
The problem with it, is that when you're using a group by clause, you cannot have columns in the select statement which do not appear inside the group by.
I think that you are on the right track, I'd suggest you though to go for an approach like this:
You need to find out how many reservations a meal has, and then compare that with the max_reservations number.
So, I would first create a query that does a left join with the reservations (this way you take into account meals that don't have even one reservation), count the reservations.meal_id and use having to return the meals ids with a higher max_reservations then the count. So, after you have the meals id you can just do a whereIn query.
I believe something close to this will do the job:
```
const availableMealsIds = knex('meal')
    .select('meal.id', 'meal.max_reservations')
    .leftJoin('reservation', { 'meal.id': 'reservation.meal_id' })
    .count({'reservations' : 'reservation.meal_id'})
    .groupBy('meal.id')
    .having(knex.raw('meal.max_reservations > reservations'));
```
And then, simply use these Ids to get the meals you need :)